### ISSUE TITLE
Parallel requests 

### ISSUE THREAD

**Issue by ciaoamigoschat** [jevakallio/redux-offline#465]
Hi I have not found anything in the documentation is it possible to execute N requests in parallel? How can I do is there any parameter in the configuration?

**Comment by sorodrigo** [comment]
At the beginning the original author had planned a batching feature, but in the end it wasn't implemented. In the upcoming release we'll add support for more powerful queue manipulation that will unlock this. For now there's no easy way to achieve it. You can try overriding the default queue.

**Comment by ciaoamigoschat** [comment]
Thanks. Do you have any suggestions / examples of how to do it?  Looking at the source code, I saw that the middleware.js processes the queue and then sends them one by one.

**Comment by sorodrigo** [comment]
You're right at the moment the send awaits for each request before picking up a new one. If you can wait a little bit longer. You can try to use the v3 beta and start using the brand new API. https://github.com/redux-offline/redux-offline/tree/v3/packages/offline-side-effects

It's almost out I swear 🙏🏻

**Comment by sorodrigo** [comment]
Or as an alternative you could try @piranna 's fork https://github.com/redux-offline/redux-offline/pull/411

**Comment by piranna** [comment]
> Or as an alternative you could try @piranna 's fork #411

My fork is intended to have multiple queues in parallel, like multiple areas in the app, but my interpretation of what @ciaoamigoschat needs, is to have multiple requests of a single app area in parallel, like `Promise.all()`. My fork would not help here, but that idea seems interesting, to have defined an inter-dependencies graph between the requests and process them in order, since not necesarily allways we need to wait to the previous request to suceed to do the next one, that's only needed when all of them operate over the same collection or resource, but if they are isolated ones, there's no problem on execute in parallel :-)

**Comment by sorodrigo** [comment]
But if you create N queues you could have them perform in parallel no?

**Comment by piranna** [comment]
> But if you create N queues you could have them perform in parallel no?

Yes, obviously, but just only one request at each time on each one of them, not multiple ones.

**Comment by jsgoupil** [comment]
I have been looking at the code for several hours to see if I would be capable of improving the library or build on top of it. This library is super extensible, but this busy state is probably the only thing stopping me. It's too baked in around the send call.
It would be beneficial to add a configuration around that.
Now the support of "multiple queues" provided here also is a bit constraining, as it basically allows to create multiple Middleware, but it still has this busy flag which prevents me from sending parallel requests easily.

This request/bug has been open a while ago, and the v3 doesn't seem to have much traction either.

I wonder what's the state of accepting smaller pull requests on this project in order to get this busy state better configurable.
I would see a configuration that could simply take the enhanced state that would return a boolean isBusy().
Otherwise. I would probably have to fork this whole thing and offer something else.

**Comment by sorodrigo** [comment]
If it's a backward-compatible change we can add it to v2. I agree v3 is taking a long time, I'm not focusing on this project at the moment so I only dedicate time occasionally to get it closer to production-ready.
